static/app/components/core/slider/index.chonk.tsx

```diff
@@ -0,0 +1,317 @@
+  const [label, setLabel] = useState(value);
+  const initialProgress = getProgress(value, min, max);
+  const filledSteps = useMemo(() => (step === -1 ? -1 : label / step), [step, label]);
+
+  const handleChange = useCallback(
+    (event: Event) => {
+      const input = event.target as HTMLInputElement;
+      const {valueAsNumber, min: nativeMin, max: nativeMax} = input;
+      setLabel(valueAsNumber);
+      const progress = getProgress(valueAsNumber, nativeMin || 0, nativeMax || 100);
+      input.parentElement?.style.setProperty('--p', `${progress.toFixed(0)}%`);
+    },
+    [setLabel]
+  );
+
+  useEffect(() => {
+    const input: HTMLInputElement | null = inputRef.current;
+    if (!input) {
+      return;
+    }
+    input.addEventListener('input', handleChange);
+    // eslint-disable-next-line consistent-return
+    return () => input.removeEventListener('input', handleChange);
+  });
+
+  return (
+    <SliderContainer
+      aria-disabled={props.disabled}
+      style={
+        {'--p': `${initialProgress.toFixed(0)}%`, '--steps': `${label}`} as CSSProperties
+      }
+    >
+      {props.step ? (
+        <SliderTicks
+          filledSteps={filledSteps}
+          n={
+            typeof props.step === 'string' ? Number.parseInt(props.step, 10) : props.step
+          }
+        />
+      ) : null}
+      <StyledSlider ref={refs} type="range" {...props} />
+      {props.disabled ? null : (
+        <SliderOutput htmlFor={props.id}>
+          <SliderLabel>{formatLabel(label)}</SliderLabel>
+        </SliderOutput>
+      )}
+    </SliderContainer>
+  );
+}
+
+function SliderTicks({
+  n,
+  filledSteps,
+}: {
+  filledSteps: number;
+  n: number;
+  disabled?: boolean;
+}) {
+  return (
+    <StepsContainer role="presentation">
+      {Array.from({length: n}, (_, i) => {
+        return <StepMark key={i} filled={i <= filledSteps} />;
+      })}
+    </StepsContainer>
+  );
+}
+
+const StepsContainer = styled('div')<{theme?: DO_NOT_USE_ChonkTheme}>`
+  pointer-events: none;
+  position: absolute;
+  bottom: 0;
+  left: 0;
+  right: 0;
+  display: flex;
+  flex-direction: row;
+  height: 14px;
+  width: calc(100% - 16px);
+  margin-inline: auto;
+
+  &::before {
+    content: '';
+    position: absolute;
+    right: 0;
+    height: 12px;
+    width: 2px;
+    border-radius: ${p => p.theme.radius.xl};
+    background: ${p => p.theme.colors.surface100};
+  }
+`;
+
+const StepMark = styled('span')<{filled?: boolean; theme?: DO_NOT_USE_ChonkTheme}>`
+  box-sizing: border-box;
+  position: relative;
+  flex-grow: 1;
+  height: 100%;
+
+  &::before {
+    content: '';
+    position: absolute;
+    height: 12px;
+    width: 2px;
+    border-radius: ${p => p.theme.radius.xl};
+    background: ${p =>
+      p.filled ? p.theme.colors.chonk.blue300 : p.theme.colors.surface100};
+  }
+`;
+
+function toNumber(value: number | string) {
+  if (typeof value === 'number') {
+    return value;
+  }
+  return Number.parseInt(value, 10);
```

Comment:
Worth checking for NaN here as well and reporting back to Sentry so we can catch it